### ISSUE TITLE
Feature: Standard management interface

### ISSUE THREAD

**Issue by ShishKabab** [WorldBrain/storex-hub#7]
This issue serves as preparation to implement:
https://www.notion.so/worldbrain/StorexHub-Interface-3a6024e9538e404ab4df41771302de3f

A rough first draft of the tasks involved (to be edited and estimated as the discussion evolves):

- Implement standard plugin directory living next to the binary
- Implement endpoint to get information about available and installed plugins
- Enabled/disable plugins?
- Serve the front-end from a certain URL
- Start the front-end on first startup
- Plugin overview UI
- Plugin settings UI
  - Standard settings
  - Links to custom settings
- App functions (index page in Memex, get tag suggestions, etc.)
- System tray
- GitHub-based plugin discovery (for now only featured plugins)
- Checking for plugin updates
- Plugin update mechanism (means downloading, overwriting, reloading.)

**Comment by ShishKabab** [comment]
Depending on the scope and budget, I'd say some things we can implement in a later phase are:
* Plugin discovery (required for 'Featured plugins')
* Plugin updates

Also, we'd open the web interface only on the first startup, right? If not, we need to implement a desktop/menu shortcut to enable the user to open the interface, as opposed to it starting without opening the interface on system startup.

The tray icon is going to take work, since there is no optimal solution yet. NW.js has crappy packaging, whereas Electron consumes way too much memory. Maybe something non-JS needs to be used here.

**Comment by blackforestboi** [comment]
> Also, we'd open the web interface only on the first startup, right? 

Yes and every time they would open it via the icon on desktop (not tray) 

> Plugin discovery (required for 'Featured plugins')

Plugin discovery will be very important for the collab partners, so should be done relatively soon after. 

Looking forward to go deeper into this list tomorrow!

**Comment by ShishKabab** [comment]
> Yes and every time they would open it via the icon on desktop (not tray)

OK, that also requires us to dive into how shortcuts work to execute the binary with different arguments. Looking at this, if it works, it should be doable:
https://www.npmjs.com/package/create-desktop-shortcuts

We do need to plan the UX around this. We don't have a Windows installer for example where we can ask the user whether they actually want a desktop shortcut. On Linux it should be enough to add a `.desktop` entry so things show in the menu, but am not sure how standardized that is. On Mac, I have no clue how things work.

**Comment by ShishKabab** [comment]
As for the systray, this looks promising:
https://www.npmjs.com/package/systray

Only consideration is how to bundle it. May be just copying another binary into the distribution.
